src/main/java/controllers/dashBoardController.java

```diff
@@ -301,31 +318,40 @@ public void handleOnSelection(ActionEvent event) {
 	public RoomModel[] toggleOnOff(String mode){
 		checkItemAndLocationSelection();
 		RoomModel[] allRoom= houseRoomsModel.getAllRoomsArray();
-		for(RoomModel rm : allRoom){
-			for(String loc: selectLocation){
-				if(!checkPermission(mainController.getLoggedUser(),loc) && rm.getName().equalsIgnoreCase(loc)) {
-					addToConsoleLog("Cannot do the command, permission denial. Trying to turn "+mode +" in "+loc);
-					continue;
-				}
-				if(rm.getName().equalsIgnoreCase(loc)){
-					switch (selectItem){
-						case "light":
-							rm.setNumOpenLights(mode.equals("on")?rm.getNumLights():0);
-							break;
-						case "door":
-							rm.setNumOpenDoor(mode.equals("on")?rm.getNumDoors():0);
-							break;
-						case "window":
-							if(!rm.isObjectBlockingWindow()){
-								rm.setNumOpenWindows(mode.equals("on")?rm.getNumWindows():0);
-							}
-							else{
-								addToConsoleLog("Cannot do the command, Object block the window of room "+loc);
-							}
-							break;
+		if(selectLocation != null) {
```

Comment:
it would not be null since method CheckItemAndLocation() is called already

Reply:
If you test selecting a door/window/light without selecting a location and clicking on open/close/auto, it will throw an exception